src/Contexts/dataReducer.js

```diff
@@ -0,0 +1,110 @@
+export const dataReducer = (state, action) => {
+  switch (action.type) {
+    case "FETCH_PRODUCTS":
+      return { ...state, products: action.payload };
+    case "FETCH_TO_WISHLIST":
+      return { ...state, wishListItems: action.payload };
+    case "SORT":
+      return { ...state, sortBy: action.payload };
```

Comment:
just a suggestion, you can extract the payload directly, you could avoid writing action throughout.

Reply:
Thanks, I have made the changes.